File: ProTip365Android/app/src/main/java/com/protip365/app/presentation/onboarding/OnboardingModels.kt
package com.protip365.app.presentation.onboarding

data class OnboardingState(
    val currentStep: Int = 0, // Start with language step (matching iOS)
    val language: String = "en",
    val useMultipleEmployers: Boolean = true,
    val singleEmployerName: String = "", // For single employer mode
    val defaultEmployerId: String? = null, // Default employer for multiple employers mode
    val employers: List<com.protip365.app.data.models.Employer> = emptyList(), // List of employers loaded from database
    val weekStart: Int = 0, // 0=Sunday, 1=Monday
    val securityType: SecurityType = SecurityType.NONE,
    val hasVariableSchedule: Boolean = false,
    val tipTargetPercentage: String = "15", // Default 15% like iOS
    val averageDeductionPercentage: String = "30", // Default 30% like iOS
    val targetSalesDaily: String = "",
    val targetSalesWeekly: String = "",
    val targetSalesMonthly: String = "",
    val targetHoursDaily: String = "",
    val targetHoursWeekly: String = "",
    val targetHoursMonthly: String = "",
    val isCompleting: Boolean = false, // Loading state for finish button
    val completionError: String? = null // Error message if completion fails
) {
    fun isStepValid(): Boolean {
        return when (currentStep) {
            0 -> true // Language selection always valid
            1 -> useMultipleEmployers || singleEmployerName.trim().isNotEmpty() // Single employer name required if not using multiple
            2 -> true // Week start always valid
            3 -> true // Security always valid
            4 -> true // Variable schedule always valid (boolean)
            5 -> tipTargetPercentage.isNotEmpty() &&
                 averageDeductionPercentage.isNotEmpty() &&
                 targetSalesDaily.isNotEmpty() &&
                 targetHoursDaily.isNotEmpty() // Core targets + deduction required
            6 -> true // How to use page always valid
            else -> false
        }
    }
    
    fun resetToDefaults() {
        // Reset all values to defaults
    }
}

enum class SecurityType {
    NONE,
    PIN,
    BIOMETRIC
}

enum class OnboardingStep(val stepNumber: Int, val titleKey: String, val descriptionKey: String) {
    LANGUAGE(0, "language_step_title", "language_step_description"),
    MULTIPLE_EMPLOYERS(1, "multiple_employers_step_title", "multiple_employers_step_description"),
    WEEK_START(2, "week_start_step_title", "week_start_step_description"),
    SECURITY(3, "security_step_title", "security_step_description"),
    VARIABLE_SCHEDULE(4, "variable_schedule_step_title", "variable_schedule_step_description"),
    TARGETS(5, "targets_step_title", "targets_step_description"),
    HOW_TO_USE(6, "how_to_use_step_title", "how_to_use_step_description")
}

File: _Backups/2025-12-08_Pre_Optimization/archive/ProTip365_backup_20250915_112051/Onboarding/OnboardingModels.swift
import SwiftUI

// MARK: - Onboarding Models

enum OnboardingStep: Int, CaseIterable {
    case language = 1
    case multipleEmployers = 2
    case weekStart = 3
    case security = 4
    case variableSchedule = 5
    case salesAndTipTargets = 6
    case howToUse = 7

    var totalSteps: Int {
        return OnboardingStep.allCases.count
    }
}

enum Field {
    case tipPercentage, dailySales, weeklySales, monthlySales, dailyHours, weeklyHours, monthlyHours
}

// SecurityType is already defined in SecurityManager.swift

// MARK: - Onboarding State

class OnboardingState: ObservableObject {
    @Published var currentStep = 1
    @Published var selectedLanguage = "en"
    @Published var useMultipleEmployers = false
    @Published var weekStartDay = 0
    @Published var selectedSecurityType = SecurityType.none
    @Published var hasVariableSchedule = false
    @Published var tipTargetPercentage = "15"
    @Published var targetSalesDaily = ""
    @Published var targetSalesWeekly = ""
    @Published var targetSalesMonthly = ""
    @Published var targetHoursDaily = ""
    @Published var targetHoursWeekly = ""
    @Published var targetHoursMonthly = ""
    @Published var showPINSetup = false
    @Published var isLoading = false
    @Published var showError = false
    @Published var errorMessage = ""

    private let totalSteps = 7
    let weekDays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]

    var isStepValid: Bool {
        switch currentStep {
        case 1: return true // Language selection always valid
        case 2: return true // Multiple employers always valid (boolean)
        case 3: return true // Week start always valid
        case 4: return true // Security always valid
        case 5: return true // Variable schedule always valid (boolean)
        case 6: return !tipTargetPercentage.isEmpty && !targetSalesDaily.isEmpty && !targetHoursDaily.isEmpty // All targets required
        case 7: return true // How to use page always valid
        default: return false
        }
    }

    func resetToDefaults() {
        currentStep = 1
        selectedLanguage = "en"
        useMultipleEmployers = false
        weekStartDay = 0
        selectedSecurityType = .none
        hasVariableSchedule = false
        tipTargetPercentage = "15"
        targetSalesDaily = ""
        targetSalesWeekly = ""
        targetSalesMonthly = ""
        targetHoursDaily = ""
        targetHoursWeekly = ""
        targetHoursMonthly = ""
        showPINSetup = false
        isLoading = false
        showError = false
        errorMessage = ""
    }
}

// MARK: - Onboarding Profile Update Model

struct OnboardingProfileUpdate: Encodable {
    let language: String
    let use_multiple_employers: Bool
    let week_start: Int
    let has_variable_schedule: Bool
    let tip_target_percentage: Double
    let target_sales_daily: Double
    let target_sales_weekly: Double
    let target_sales_monthly: Double
    let target_hours_daily: Double
    let target_hours_weekly: Double
    let target_hours_monthly: Double
}